### DISCUSSION TITLE
Delays when activating HASS scene

### DISCUSSION THREAD

**Discussion by cweakland** [zwave-js/zwave-js#2181] [Q&A]
First of all, thanks for everyone's work on this project, keep up the great work!  

I have a 32 node zwave network that I recently moved from the deprecated zwave integration to zwavejs. Individually everything works properly. The issue is that I see delays when activating scenes in HASS that contain several lights/switches. The delays are usually around 10 seconds in duration, on occasion I have seen where some dimmer levels are not set correctly (not captured in this log). Also, it should be noted that I do not have this issue with the deprecated zwave integration.

In most cases I see the following error around the time of the delay: "Timed out while waiting for a response from the node" In the attached log here are the times of the delays observed when activating a scene in HASS:

  These were general heavy delays during these times, these lasted more than 10 seconds:
  11:13:51.22 - 11:14:06.20
  11:14:55.83 - 11:15:01.95

  Here I noted some specific node delays:
  Node 040 - Light off (delayed): 11:15:43.57
  Node 040 - Light on (delayed): 11:17:45.83

Again, this is not isolated to a given node, the issue seems to move around from node to node. I am not sure if I have a specific bug so I am hesitant to open a bug until I can isolate the issue, any help would be appreciated. 

[zwavejs_368.log](https://github.com/zwave-js/node-zwave-js/files/6217574/zwavejs_368.log)

Thanks!

**Comment by AlCalzone** [comment]
It looks like you have several nodes that zwave-js tries to control with Supervision, but the device does not respond to that command (probably unsupported). These are:
* Node 29

Some others report no support which zwave-js doesn't seem to pick up because they are received out of sequence:
* Node 26
* Node 40
* Node 41

We'll need to fix that, but we should get around both by disabling supervision with a compat flag. Which devices are that (manufacturer ID, product type, product ID, firmware version)?

I also see that some of your devices don't respond to the query for some config params, which we need to change in our config files:
* Node 26 (7, 8)
* Node 27 (7, 8)
* Node 29 (7, 8)
* Node 39 (7, 8)
* Node 40 (7, 8)
* Node 41 (7, 8)
* Node 51 (132, 137, 185)
* Node 58 (9, 10)
* Node 38 (2, 250)

I'm guessing the first few are the same device - which devices are these (manufacturer ID, product type, product ID, firmware version)?

**Reply by cweakland** [reply]
Thanks a bunch for taking a look at my issue, I really appreciate it!  

To answer your questions here is the node information requested:

    Node 26 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 27 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 29 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 39 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 40 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 41 (7, 8) - Zooz ZEN22 Dimmer v2, Manuaf. ID 634, Prod Type: 45330, Prod. ID 7964,  Firmware 1.7
    Node 51 (132, 137, 185) - Trane XR524 Thermostat,  Manuaf. ID 139, Prod Type: 21586, Prod. ID 21570,  Firmware 1.30
    Node 58 (9, 10) - Zooz ZEN21 Dimmer v2, Manuaf. ID 634, Prod Type: 45329, Prod. ID 7708,  Firmware 3.0
    Node 38 (2, 250) - AEON Labs Micro Switch G2 DSC26103-ZW, Manuaf. ID 134, Prod Type: 3, Prod. ID 26,  Firmware 2.14

**Reply by cweakland** [reply]
Also, I have some email correspondence with Zooz going, I let them know about this thread. I may not be a programmer, but if there is any legwork I can do let me know.

**Reply by cweakland** [reply]
Zooz got back to me and they have much newer firmware (ZEN22_V403) for my devices. Let me upgrade my devices and go from there. Additionally, doing the upgrade within the zwavejs2mqtt console was amazing, replacing the device after the upgrade, also amazing! :D

For anyone else that goes down this road, the firmware update takes time you get a small progress meter in the upper right of the console. Afterwards the node goes dead, you then want to choose "replace" for the failed node and put the switch into inclusion mode. 

Edit: put back my original comment

**Reply by AlCalzone** [reply]
No you usually don't want to replace the node. Give it some time to load the firmware.

**Reply by cweakland** [reply]
Its funny some nodes do this okay, some just go dead. Either way it works. Thanks again!

**Reply by AlCalzone** [reply]
Worth a look into the user manual what is said about that. Maybe they need to be power cycled after a while or so 🤷🏻‍♂️.

**Reply by cweakland** [reply]
I'm back, I upgraded the firmware on all my Zooz nodes (with the exception of one older node). I have not seen any any misconfiguration events (i.e. the lights get set the way the scene instructs it), however, I still see some 10 second delays.  Here is the latest log and two of the times I noticed the delay were:

22:20:15.13
22:21:28.63

[zwavejs_355.log](https://github.com/zwave-js/node-zwave-js/files/6239911/zwavejs_355.log)

Thanks again for talking a look. Let me know if you want me to try anything else?

**Comment by cweakland** [comment]
Thanks! I did just this and it worked like a charm.
